src/AlbyOnContainers.IdentityServer/Controllers/RolesController.cs

```diff
@@ -0,0 +1,81 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using IdentityServer.Infrastructure;
+using IdentityServer.Models;
+using IdentityServer.Models.RolesViewModels;
+
+namespace Roles.Controllers
+{
+    [Authorize(Roles = "Admin")]
+    public class RolesController : Controller
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly ApplicationDbContext _context;
+        private readonly RoleManager<IdentityRole> _roleManager;
+
+        public RolesController(
+            ApplicationDbContext context,
+            UserManager<ApplicationUser> userManager,
+            RoleManager<IdentityRole> roleManager)
+        {
+            _roleManager = roleManager;
+            _userManager = userManager;
+            _context = context;
```

Comment:
You don't need to inject the dbContext use role and user manger collections.

Reply:
_context is needed to get the UserRoles